django_mailbox/models.py

```diff
@@ -654,10 +658,16 @@ def set_body(self, body):
         no fields existed for storing arbitrary bytes.
 
         """
-        if six.PY3:
+        settings = utils.get_settings()
+
+        if six.PY3 and isinstance(body, six.text_type):
             body = body.encode('utf-8')
+
         self.encoded = True
-        self.body = base64.b64encode(body).decode('ascii')
+        self.body = base64.b64encode(body).decode(
+            settings['default_charset'],
+            'replace',
```

Comment:
Nor here; base64 encoding shouldn't have to worry about default_charset at all.  I was just a little overzealous in removing those ascii decode/encodes.